src/mnemonics/electrum-words.cpp

```diff
@@ -159,6 +166,7 @@ namespace
         return true;
       }
       // Some didn't match. Clear the index array.
+      memwipe(matched_indices.data(), matched_indices.size() * sizeof(matched_indices[0]));
```

Comment:
I ran the unit test and found it crash here due to `matched_indices` being empty. I think for every call to `memwipe` against a dynamically allocated buffer, its size needs to be checked.

Reply:
https://github.com/monero-project/monero/pull/4126

Reply:
That's not enough because I get `Error: memset_s failed`.

Reply:
Then logs whatever inputs it gets. If it sets errno, print errno too.

Reply:
Looks like its return value is an errno value.

Reply:
Why don't you simply add this test `if (!matched_indices.empty())` here? `matched_indices` would be empty if the first seed word is not found in the first item in `language_instances`, and it would be logically wrong to call `memwipe` for an empty buffer.

Reply:
Because it seems better to do it once only. It is also equivalent, no ? Except for integer overflow, but that should not happen here unless the size field of the vector is corrupted.

Reply:
OK, but then shouldn't the same check be done for all the 3 versions of `memwipe` in memwipe.c? #4126 adds it to just one of them.

Reply:
Oh you're right, somehow I thought the one at the bottom was the entry for the others. Fixing...

Reply:
OK then :)
The patch could've been put into this PR, but OK as long as #4126 and this get merged together.